internal/controllers/clustertemplate_controller.go

```diff
@@ -239,6 +253,69 @@ func validateConfigmapReference[T any](
 	return nil
 }
 
+// validateName return true if the ClusterTemplate name is the
+// format: <name>.<version>, false otherwise
+func validateName(c client.Client, name, version, metadataName, namespace string) error {
+	if metadataName != name+"."+version {
+		return utils.NewInputError("failed to validate ClusterTemplate name %s, should be in the format <spec.name>.<spec.version>: %s", metadataName, name+"."+version)
+	}
+
+	allClusterTemplates := &oranv1alpha1.ClusterTemplateList{}
+	err := c.List(context.Background(), allClusterTemplates)
+	if err != nil {
+		return fmt.Errorf("could not get list of ClusterTemplate across the cluster: %w", err)
+	}
+
+	sameNameVersion := map[string]bool{}
+	sameMetadataName := map[string]bool{}
+	for _, aClusterTemplate := range allClusterTemplates.Items {
+		if aClusterTemplate.Namespace == namespace {
+			continue
+		}
+		if aClusterTemplate.Name == metadataName {
+			sameMetadataName[aClusterTemplate.Namespace] = true
+		}
+		if aClusterTemplate.Spec.Name == name &&
+			aClusterTemplate.Spec.Version == version {
+			sameNameVersion[aClusterTemplate.Namespace] = true
+		}
+	}
+	if len(sameMetadataName) != 0 {
+		return utils.NewInputError("failed to validate ClusterTemplate name %s, a identical name already exists in namespaces: %s",
+			metadataName, sliceToString(mapKeysToSlice(sameMetadataName)))
+	}
+	if len(sameMetadataName) != 0 {
+		return utils.NewInputError("failed to validate ClusterTemplate name %s, the combination of <spec.name>.<spec.version>: %s already exists in namespace: %s",
+			metadataName, name+"."+version, sliceToString(mapKeysToSlice(sameNameVersion)))
+	}
+	return nil
+}
+
+// validateTemplateID return true if the templateID is a valid uuid, false otherwise
+func validateTemplateID(ctx context.Context, c client.Client, object *oranv1alpha1.ClusterTemplate) error {
+	if object.Spec.TemplateID != "" {
+		_, err := uuid.Parse(object.Spec.TemplateID)
+		if err != nil {
+			return utils.NewInputError("failed to validate templateID, invalid UUID:%s", object.Spec.TemplateID)
+		}
+		return nil
+	}
+
+	newID := uuid.New()
+	newTemplate := object
```

Comment:
I think this should be
newTemplate := object.DeepCopy()

Reply:
Fixed